### PR TITLE
Fix up Auto infrastructure

### PR DESCRIPTION
**Pull request by prensing**
Lots of changes:
- add SmartDashboard selector for start pose
- fix start poses
- need to use DeferredCommand for the "initialPose" because it is selected at start of auto
- remove use of AutoCommandInterface; not needed since initialPose is separate from Auto routine

### CODE REVIEW

File: src/main/java/frc/robot/FieldConstants.java [hunk 1]
```diff
@@ -50,24 +49,29 @@ public class FieldConstants {
...
+    // Java magic: use a static code block to set START_1 and _3 angle to point exactly 
+    //    at the SPEAKER.
+    static {
+        Translation2d pos = new Translation2d(1.25, 3.85);
+        ROBOT_START_1 = new Pose2d(pos, SPEAKER.minus(pos).getAngle());
 
-    public static boolean isRedAlliance() {
-        Optional<Alliance> alliance = DriverStation.getAlliance();
-        return alliance.isPresent() && alliance.get() == Alliance.Red;
+        pos = new Translation2d(1.25, 6.95);
```
Comment (jacobh1) [inline src/main/java/frc/robot/FieldConstants.java:62]: This looks like a constant that could be a named point?

**Review by jacobh1 — APPROVED** [review]

File: src/main/java/frc/robot/FieldConstants.java (hunk 1 above)
Comment (prensing) [reply src/main/java/frc/robot/FieldConstants.java:62]: This was just a quick, accurate way to compute the angle. I don't think we need a separate static Translation2d for just the start position.

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -128,10 +138,23 @@ private void configureAutos() {
...
-    public AutoCommandInterface getAutonomousCommand() {
+    public Pose2d getInitialPose() {
+        return FieldConstants.flipPose(m_startLocation.getSelected());
+    }
+
+    public Command getAutonomousCommand() {
         return m_chosenAuto.getSelected();
     }
 
+    public boolean autoHasChanged() {
```
[the remark this answers is not in the document]
Comment (prensing) [reply src/main/java/frc/robot/RobotContainer.java:149]: It is only tracking the selection in the Chooser. If/when we get to dynamic construction of an Auto from a string, we will have to enhance this.

File: src/main/java/frc/robot/commands/GetNoteX.java [hunk 3]
```diff
@@ -72,14 +74,14 @@ public GetNoteX(Translation2d targetNote, DriveTrain driveTrain, NoteVision note
...
 
         if (FieldConstants.isCenterNote(targetNote)) {
             // Use note "monitoring" for center notes only
-            addCommands(m_driveTrain.followPath(() -> getInitialPath())
+            addCommands(new DeferredCommand(() -> m_driveTrain.followPath(getInitialPath()), Set.of(m_driveTrain))
                         .alongWith(new MonitorForNote(noteVision, () -> m_driveTrain.getPose(), m_targetNote, this)));
         } else {
-            addCommands(m_driveTrain.followPath(() -> getInitialPath()));
+            addCommands(new DeferredCommand(() -> m_driveTrain.followPath(getInitialPath()), Set.of(m_driveTrain)));
         }
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/commands/GetNoteX.java:85]: Why are we only doing MonitorForNote for CenterNotes?

File: src/main/java/frc/robot/commands/GetNoteX.java [hunk 4]
```diff
@@ -89,15 +91,10 @@ public GetNoteX(Translation2d targetNote, DriveTrain driveTrain, NoteVision note
...
-        return FieldConstants.flipPose(firstPath.getStartingDifferentialPose());
-    };
-
     private PathPlannerPath getInitialPath() {
         Pose2d pose = m_driveTrain.getPose();
         Pose2d poseBlue = FieldConstants.flipPose(pose);
-
+        System.out.println("Starting getInitialPath " + poseBlue);
         // this part used when in center note area, if intended center note is not found
         if (poseBlue.getX() > FieldConstants.BLUE_WING_LINE_X_METERS) {
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/commands/GetNoteX.java:99]: Doesn't this value have to get flipped if we're Red?

File: src/main/java/frc/robot/commands/GetNoteX.java (hunk 3 above)
Comment (prensing) [reply src/main/java/frc/robot/commands/GetNoteX.java:85]: It is just too close for the Stage notes. The algorithm (as written) looks over a distance window which won't work over short distances. Seem low risk of our team mate stealing the notes. When we get field experience, we could add this if we want.

File: src/main/java/frc/robot/commands/GetNoteX.java (hunk 4 above)
Comment (prensing) [reply src/main/java/frc/robot/commands/GetNoteX.java:99]: "pose" is the current real pose. If we are Red, the flipPose call will flip it, and not if we are Blue, so "poseBlue" is on the Blue side. Then we can just compare with Blue landmarks.

**Review by coachFitz0807 — APPROVED** [review]

**Pull request merged**